eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/ListIterable.java

```diff
@@ -303,4 +304,22 @@ default int binarySearch(T key)
      * @since 6.0
      */
     ListIterable<T> subList(int fromIndex, int toIndex);
+
+    default <T2> void forEachInBoth(ListIterable<T2> other, Procedure2<? super T, ? super T2> procedure)
+    {
+        if (other != null)
+        {
+            if (this.size() == other.size())
+            {
+                this.forEachWithIndex((each, index) -> procedure.value(each, other.get(index)));
+            }
+            else
+            {
+                throw new IllegalArgumentException("Attempt to call forEachInBoth with two Lists of different sizes :"
+                        + this.size()
+                        + ':'
+                        + other.size());
+            }
+        }
```

Comment:
The null check is uncovered by tests. Isn't it a bug? I can pass in forEachInBoth(null, procedure) and it silently does nothing?

Reply:
Yes, I'll add `Objects.requireNonNull(other)`.